src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -925,9 +960,25 @@ def common_task_schema(self, provider):
         return schema
 
 
+async def devd_zfs_hook(middleware, data):
+    if data.get('type') == 'misc.fs.zfs.pool_import':
+        asyncio.ensure_future(middleware.call('cloudsync.credentials.migrate_sftp_key_file'))
+
+
+async def system_event(middleware, event_type, args):
+    if args['id'] == 'ready':
+        asyncio.ensure_future(middleware.call('cloudsync.credentials.migrate_sftp_key_file'))
+
+
 async def setup(middleware):
     for module in load_modules(os.path.join(os.path.dirname(os.path.realpath(__file__)), os.path.pardir,
                                             "rclone", "remote")):
         for cls in load_classes(module, BaseRcloneRemote, []):
             remote = cls(middleware)
             REMOTES[remote.name] = remote
+
+    for credential in await middleware.call("cloudsync.credentials.query", [["provider", "=", "SFTP"]]):
```

Comment:
I am still puzzled why we cant have a one time migration instead of having to support this code forever in middlewared code?

Reply:
Because these keys are stored on datasets and datasets are not mounted yet on migrations stage.

Reply:
But I didnt say database migration, but one time script migrations.

Its not the first time we have needed this, and I doubt its gonna be the last

Reply:
Is there any example of such script already present? Couldn't find any.

Reply:
No, but thats my point.

We have done many ad-hocs solutions like this for a long time, varying from ix rc.d scripts that would set sentinel files in /data to not run again and permanent code to handle migration.

Perhaps its time we come up with a better solution for such cases